src/apps/activities/services/activity_history.py

```diff
@@ -0,0 +1,136 @@
+                    )
+                )
+            elif new_activity and prev_activity:
+                change = ActivityHistoryChange()
+                has_changes = False
+                if prev_activity.name != new_activity.name:
+                    has_changes = True
+                    if generator.is_considered_empty(new_activity.name):
+                        change.name = generator.cleared_text("name")
+                    else:
+                        change.name = generator.filled_text(
+                            "name", new_activity.name
+                        )
+                if prev_activity.description != new_activity.description:
+                    has_changes = True
+                    change.description = generator.updated_text("description")
+                if prev_activity.splash_screen != new_activity.splash_screen:
+                    has_changes = True
+                    change.splash_screen = generator.updated_text(
+                        "splash screen"
+                    )
+                if prev_activity.image != new_activity.image:
+                    has_changes = True
+                    change.image = generator.updated_text("image")
+                if (
+                    prev_activity.show_all_at_once
+                    != new_activity.show_all_at_once
+                ):
+                    has_changes = True
+                    change.show_all_at_once = generator.changed_text(
+                        prev_activity.show_all_at_once,
+                        new_activity.show_all_at_once,
+                    )
+                if prev_activity.is_skippable != new_activity.is_skippable:
+                    has_changes = True
+                    change.is_skippable = generator.changed_text(
+                        prev_activity.is_skippable, new_activity.is_skippable
+                    )
+                if prev_activity.is_reviewable != new_activity.is_reviewable:
+                    has_changes = True
+                    change.is_reviewable = generator.changed_text(
+                        prev_activity.is_reviewable, new_activity.is_reviewable
+                    )
+                if (
+                    prev_activity.response_is_editable
+                    != new_activity.response_is_editable
+                ):
+                    has_changes = True
+                    change.response_is_editable = generator.changed_text(
+                        prev_activity.response_is_editable,
+                        new_activity.response_is_editable,
+                    )
+                if prev_activity.ordering != new_activity.ordering:
+                    has_changes = True
+                    change.ordering = generator.changed_text(
+                        prev_activity.ordering, new_activity.ordering
+                    )
+                if has_changes:
+                    activity_changes.append(change)
+        return activity_changes
```

Comment:
1. Don't you think that `generator` is a good naming for any kind of variable name. Could you specify it more (For what this generator is created. what does it implement)
2. Don't you think this huge part of `is` statements would be moved somewhere else?
3. Also, I think it would be super helpful if you would leave comments on each `if` statement. This part of the code is huge and it takes much time to figure out what is going on.

Reply:
As a suggestion, you can encapsulate the `generator` inside `ActivityHistoryChange` as far as nobody uses it.

Reply:
1. I wanted to use `changes_text_generator`, but it seemed to me weird, so then changed to generator, if you have and suggestion pls write
2. ?
3. do you think it needs comments/docstring, I think no, it is doing what is written on the name of method

Reply:
`ChangeTextGenerator` is also used in AppletHistoryChange

Reply:
1. You don't need to use that weird `_generator` postfix. We create classes but we do not call them `PersonClass` for example, right? You can easily use `text_changer: Generator = ...`
2. typo. It was about a huge `if` statements part, that I think should be moved out
3. All parts that declare some business logic should be described using business language like what it does from the business perspective. It is not clear what idea are you trying to represent using your code. Another thing, tell me how developers should check your PR if the [story in Jira](https://mindlogger.atlassian.net/browse/M2-396) does not have any description, we don't have any overall technical task, and also, you didn't leave any description regarding the business idea. It means no developers can validate your code and that means that there is no sense of code review in general.
4. I saw that you used id in a couple of places. Even there you use different naming for some reason. (check the image)

![Alacritty 2023-01-24](https://user-images.githubusercontent.com/117179957/214291944-0f4d9f3b-a994-4f77-9a46-456cb29735c6.png)


Then, you can resolve it for free as far as we don't have any requirements according to that feature. I don't care actually :)